qt/widgets/instrumentview/src/BankRenderingHelpers.cpp

```diff
@@ -0,0 +1,153 @@
+  pos += Mantid::Kernel::V3D(xstep * (+0.5), ystep * (-0.5),
+                             0.0); // Adjust to account for the size of a pixel
+  glVertex3f(static_cast<GLfloat>(pos.X()), static_cast<GLfloat>(pos.Y()),
+             static_cast<GLfloat>(pos.Z()));
+}
+
+void extractHexahedron(const Mantid::Geometry::IObject &shape,
+                       std::vector<Mantid::Kernel::V3D> &hex) {
+  const auto &shapeInfo = shape.getGeometryHandler()->shapeInfo();
+  const auto &points = shapeInfo.points();
+  hex.assign(points.begin(), points.begin() + 4);
+}
+
+void rotateHexahedron(std::vector<Mantid::Kernel::V3D> &hex,
+                      const Mantid::Kernel::Quat &rotation) {
+  for (auto &pos : hex)
+    rotation.rotate(pos);
+}
+
+void offsetHexahedronPosition(std::vector<Mantid::Kernel::V3D> &hex,
+                              const Mantid::Kernel::V3D &offset) {
+  for (auto &pos : hex)
+    pos += offset;
+}
+
+} // namespace
+
+namespace MantidQt {
+namespace MantidWidgets {
+namespace BankRenderingHelpers {
+
+std::pair<size_t, size_t> getCorrectedTextureSize(const size_t width,
+                                                  const size_t height) {
+  return {roundToNearestPowerOfTwo(width), roundToNearestPowerOfTwo(height)};
+}
+
+void renderRectangularBank(const Mantid::Geometry::ComponentInfo &compInfo,
+                           size_t index) {
+  auto bank = compInfo.quadrilateralComponent(index);
+  auto xstep = (compInfo.position(bank.bottomRight).X() -
+                compInfo.position(bank.bottomLeft).X()) /
+               static_cast<double>(bank.nX);
+  auto ystep = (compInfo.position(bank.topRight).Y() -
+                compInfo.position(bank.bottomLeft).Y()) /
+               static_cast<double>(bank.nY);
+  // Because texture colours are combined with the geometry colour
+  // make sure the current colour is white
+  glColor3f(1.0f, 1.0f, 1.0f);
+
+  // Nearest-neighbor scaling
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glEnable(GL_TEXTURE_2D); // enable texture mapping
+
+  size_t texx, texy;
+  auto res = getCorrectedTextureSize(bank.nX, bank.nY);
+  texx = res.first;
+  texy = res.second;
+  double tex_frac_x = static_cast<double>(bank.nX) / static_cast<double>(texx);
+  double tex_frac_y = static_cast<double>(bank.nY) / static_cast<double>(texy);
+
+  glBegin(GL_QUADS);
+
+  auto basePos = compInfo.position(bank.bottomLeft);
+
+  glTexCoord2f(0.0, 0.0);
+  auto pos = compInfo.position(bank.bottomLeft) - basePos;
+  addVertex(pos, xstep, ystep);
+
+  glTexCoord2f(static_cast<GLfloat>(tex_frac_x), 0.0);
+  pos = compInfo.position(bank.bottomRight) - basePos;
+  addVertex(pos, xstep, ystep);
+
+  glTexCoord2f(static_cast<GLfloat>(tex_frac_x),
+               static_cast<GLfloat>(tex_frac_y));
+  pos = compInfo.position(bank.topRight) - basePos;
+  addVertex(pos, xstep, ystep);
```

Comment:
If/when position argument to `addVertex` is made const, it would save some lines of code just to pass the `compInfo.position(...` directly rather than the multiple reassignments going on here.